src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -659,6 +657,42 @@ static void AnimateRootView(CGRect rect)
 			ContainerView.Frame = rect;
 	}
 
+	static void ApplyContentInset(UIScrollView? scrolledView, UIScrollView? lastScrollView)
+	{
+		if (scrolledView is null || lastScrollView is null || ContainerView is null)
+			return;
+
+		var keyboardIntersect = CGRect.Intersect(KeyboardFrame, scrolledView.Frame);
+		nfloat movedContainerDistance = 0;
+
+		// if we are moving or have moved the containerview frame, we need to account for that
+		if (TopViewBeginOrigin != InvalidPoint && ContainerView is not null)
+		{
+			movedContainerDistance = TopViewBeginOrigin.Y - ContainerView.Frame.Y;
+		}
+
+		var frameInWindow = ContainerView!.ConvertRectToView(scrolledView.Frame, null);
```

Comment:
Should this be the following?
```cs
var frameInWindow = ContainerView!.ConvertRectFromView(scrolledView.Frame, scrolledView.Superview);
```
I guess we want to evaluate the `.Frame` rect from the superview coordinate system.